src/main/java/io/cdap/plugin/servicenow/connector/ServiceNowConnector.java

```diff
@@ -0,0 +1,167 @@
+import io.cdap.plugin.servicenow.util.SourceQueryMode;
+import io.cdap.plugin.servicenow.util.SourceValueType;
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+
+
+/**
+ * ServiceNow Connector Plugin
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(ServiceNowConnector.NAME)
+@Description("Connection to access data in Servicenow tables.")
+public class ServiceNowConnector implements Connector {
+  public static final String NAME = "ServiceNow";
+  private static final Logger LOG = LoggerFactory.getLogger(ServiceNowConnector.class);
+  private static final String OBJECT_TABLE_LIST = "sys_db_object";
+  private static final String ENTITY_TYPE_TABLE = "table";
+  private static final String LABEL_NAME = "label";
+  private static final String OAUTH_URL_TEMPLATE = "%s/oauth_token.do";
+
+  private final ServiceNowConnectorConfig config;
+
+  ServiceNowConnector(ServiceNowConnectorConfig config) {
+    this.config = config;
+
+  }
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateCredentials(collector);
+  }
+
+  @Override
+  public BrowseDetail browse(ConnectorContext connectorContext, BrowseRequest browseRequest) throws IOException {
+    int count = 0;
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateCredentials(collector);
+    collector.getOrThrowException();
+    BrowseDetail.Builder browseDetailBuilder = BrowseDetail.builder();
+    TableList tables = listTables();
+    Table[] table = tables.getResult();
+    for (int i = 0; i < table.length; i++) {
+      String name = table[i].getName();
+      String label = table[i].getLabel();
+      BrowseEntity.Builder entity = (BrowseEntity.builder(name, name, ENTITY_TYPE_TABLE).
+        canBrowse(false).canSample(true));
+      entity.addProperty(LABEL_NAME, BrowseEntityPropertyValue.builder(label, BrowseEntityPropertyValue.
+        PropertyType.STRING).build());
+      browseDetailBuilder.addEntity(entity.build());
+      count++;
+    }
+    return browseDetailBuilder.setTotalCount(count).build();
+  }
+
+  public TableList listTables() {
+    ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+      config.getRestApiEndpoint(), OBJECT_TABLE_LIST);
+    String accessToken = null;
+    ServiceNowTableAPIClientImpl serviceNowTableAPIClient = new ServiceNowTableAPIClientImpl(config);
+    try {
+      accessToken = serviceNowTableAPIClient.getAccessToken();
+    } catch (OAuthSystemException | OAuthProblemException e) {
+      e.printStackTrace();
```

Comment:
remove e.printStackTrace and wrap exceptions in IOException and throw.

Reply:
remove printstackTrace and write like
throw new IOException(e);

Reply:
Done